### ISSUE TITLE
Multimesh rotation about individual origins

### ISSUE THREAD

**Issue by lukostello** [godotengine/godot-proposals#2693]
### Describe the project you are working on
I'm working on a sokoban-like with twisty puzzle elements. 

### Describe the problem or limitation you are having in your project
I am using the multimesh to make the illusion that there is an infinite repeating set of the level. The twisty puzzle elements of my levels rotate and when rotating the multimesh object it rotates all the other objects about the center of the co-ordinates of the multimesh array. While this there is a rational to this which is consistent and intuitive, I would like to make the case that rather than that being the default behavior each mesh should rotate about their individual origins.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Make the default rotational behavior of multimeshes to rotate about individual origins or at least have a boolean to toggle between the two different modes. 

### If this enhancement will not be used often, can it be worked around with a few lines of script?
currently the work around is to use a vertex shader and feed in a quaternion or possibly rigging the mesh to a bone and manipulating the bone. 

### Is there a reason why this should be core and not an add-on in the asset library?
2 main reasons. First being I think there are more use cases where we would want the meshes to rotate about individual origins. One example that comes to mind is coins that rotate in place. You wouldn't want all the coins to rotate about one particular coin. and 2. the work around for if this was the default behavior is much easier to overcome than overcoming its current behavior; you just make it the child of a spatial node then that rotates all the other meshes.

**Comment by clayjohn** [comment]
To rotate individual instances in the multimesh you use the function ``set_instance_transform()``

For example to rotate a bunch of coins around the y-axis you would write:
```
for coin in range(multimesh.instance_count):
    multimesh.set_instance_transform(coin, Transform().rotated(Vector3(0, 1, 0), rotation_amount))
```

**Comment by lukostello** [comment]
I'm under the impressing that doing that method sends more instructions to the CPU negating the advantages of the multimesh. Hence why I elected for using a shader instead.

**Comment by lukostello** [comment]
I just realized that if this was implemented, then making it the child of a spatial node wouldn't accomplish the same effect as rotating the way it rotates now. because if your rotated the parent it would just rotate the child which would then rotate about individual origins. So it wouldn't work. I'll just have to make a tutorial on how I was able to accomplish the same effect with shaders for anyone who is interested.
